test/kubernetes/testutils/assertions/assertion.go

```diff
@@ -0,0 +1,8 @@
+package assertions
+
+import "context"
+
+// ClusterAssertion is a function which asserts a given behavior at a point in time
+// If it succeeds, it will not return anything
+// If it fails, it will panic
```

Comment:
If a user is defining a custom assertion that doesn't utilize testing frameworks that exist, this should be more strongly worded to indicate that the only valid failure mode is a panic (as opposed to, say a `log.Fatal`)

```suggestion
// If it fails, it must panic
```